server/src/main/java/io/crate/analyze/expressions/ExpressionAnalyzer.java

```diff
@@ -459,13 +460,20 @@ protected Symbol visitExpression(Expression node, ExpressionAnalysisContext cont
 
         @Override
         protected Symbol visitCurrentTime(CurrentTime node, ExpressionAnalysisContext context) {
-            if (!node.getType().equals(CurrentTime.Type.TIMESTAMP)) {
+            CurrentTime.Type type = node.getType();
+            if (!(type.equals(CurrentTime.Type.TIMESTAMP) || type.equals(CurrentTime.Type.TIME))) {
                 visitExpression(node, context);
             }
+            String funcName = CurrentTimestampFunction.NAME;
+            int defaultPrecision = CurrentTimestampFunction.DEFAULT_PRECISION;
+            if (type.equals(CurrentTime.Type.TIME)) {
+                funcName = CurrentTimeFunction.NAME;
+                defaultPrecision = CurrentTimeFunction.DEFAULT_PRECISION;
+            }
```

Comment:
I think a single conditional would make this a bit easier to parse. Maybe with something like this?

```java
String funcName;
int defaultPrecision;
switch (type) {
    case CurrentTime.Type.TIMESTAMP:
        funcName = ...
        break;

    case CurrentTime.Type.TIME:
        funcName = ...
        break;


    default:
      return visitExpression(node, context);
}
return allocateFunction(...
```


I think it would also be an option to make the default precision argument optional and keep the "what value is the default" knowledge encapsulated within the function itself. So that the expression analyzer doesn't add the argument if it is not specified.